### PR TITLE
Add default network setting

### PR DESCRIPTION
**Pull request by guangbochen**
Add default network setting

### CODE REVIEW

File: pkg/network/bridge_vlan_controller.go [hunk 1]
```diff
@@ -33,18 +30,25 @@ func Register(ctx context.Context, apply apply.Apply, setting harvcontroller.Set
 		apply:         apply,
 	}
 
+	err := initNetworkSettings(controller.settingClient)
+	if err != nil {
```
Comment (gitlawr) [inline pkg/network/bridge_vlan_controller.go:34]: Nit:
```
if err:=f();err!=nil{
}
```

File: pkg/network/settings.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NetworkSetting struct {
+	// physical NIC(network interface card)
+	NIC string
+
+	// previous configured NIC
+	PreviousNIC string
```
Comment (gitlawr) [inline pkg/network/settings.go:18]: The former `ConfiguredNIC` sounds better than `PreviousNIC`

File: pkg/network/settings.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+func initNetworkSettings(settingClient harvcontroller.SettingClient) error {
+	_, err := settingClient.Get(networkSettingsName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
```
Comment (gitlawr) [inline pkg/network/settings.go:24]: Errors other than not-found-error are omitted.

File: pkg/network/settings.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+				return err
+			}
+			setting := &settingsv1.Setting{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: networkSettingsName,
+				},
+				Default: string(jsonNetwork),
+			}
+			sett, err := settingClient.Create(setting)
+			if err != nil {
```
Comment (gitlawr) [inline pkg/network/settings.go:37]: Handle is-already-exist error

**Comment by guangbochen** [comment]
@gitlawr updated, thanks.

File: pkg/network/settings.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+				if errors.IsAlreadyExists(err) {
+					logrus.Println("skip to create the default network setting as it is already exist")
+				}
+				return err
```
Comment (gitlawr) [inline pkg/network/settings.go:41]: We still return the IsAlreadyExists error

Comment (guangbochen) [reply pkg/network/settings.go:44]: updated

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**